### ISSUE TITLE
Error on first use

### ISSUE THREAD

**Issue by malleeblue** [HabeltWi/AutoForm#1]
Hi,
This plugin is a great idea. I installed it but get this message when I try to run it.

2017-02-10T08:23:40	1	Traceback (most recent call last):
			  File "C:/Users/xxxxxxxxx/.qgis2/python/plugins\AutoForm\autoform.py", line 62, in handleFormofLayer
			    self.identifyRelations(selected_layer)
			  File "C:/Users/xxxxxxxxx/.qgis2/python/plugins\AutoForm\autoform.py", line 116, in identifyRelations
			    cur = self.connector.uriDatabaseConnect(uri)
			  File "C:/Users/xxxxxxxxx/.qgis2/python/plugins\AutoForm\connector.py", line 32, in uriDatabaseConnect
			    QMessageBox.warning(self.iface.mainWindow(), "Connection Error", "Failed to connect to database. Please make sure that your connection information is correct.")
			AttributeError: Connector instance has no attribute 'iface'

Any ideas of what this means and why it may be happening?

Johanna

**Comment by HabeltWi** [comment]
Hello maleechick

It turns out I made a mistake in the code and forgot to properly pass the iface variable to the connector instance. Presumably there was a problem with your connection variables and when the program tried to display the message box to tell you this, it crashed. This problem should be fixed now, but there may still be something wrong with the connection. I have pushed a newer version which you can try out instead.

William

**Comment by malleeblue** [comment]
Thanks for the feedback. I have updated my installation of the plugin and now I get no error. BUT, perhaps my expectations are not what they should be. When I run the plugin I simply get a message box that says "Form widgets were successfully changed".
I thought that I was going to get a form that I could use to change attributes of the data in the selected layer. I don't know what 'form widgets' are or which ones have been changed.
Johanna
